Rocky Linux 9.7 packages

elelaysh opened the pull request:
- upgrade stackhpc/2025.1 to Rocky Linux 9.7
- DOCA 3.2.1 for Rocky Linux 9.7
- all packages upgrade (docker, grafana, ...)

ansible/maintenance/rocky-97-ofed-upgrade.yml is an upgrade playbook
for sites using a local pulp server and doca.

See #2061 for container and overcloud host upgrade

elelaysh committed:
Switch to Rocky Linux 9.7 and DOCA 3.2.1

- DOCA 3.2.1 for RL 9.7
- Bump Rocky 9 Security SIG repo, add source

cityofships commented on etc/kayobe/pulp-repo-versions.yml:
```diff
@@ -18,6 +18,7 @@ stackhpc_pulp_repo_doca_2_9_1_rhel9_5_version: 20241211T171301
 stackhpc_pulp_repo_doca_2_9_1_rhel9_5_modules_version: 20250115T150314
 stackhpc_pulp_repo_doca_2_9_3_rhel9_6_version: 20250703T135021
 stackhpc_pulp_repo_doca_2_9_3_rhel9_6_modules_version: 20250714T141841
+stackhpc_pulp_repo_doca_3_2_0_rhel9_version: 20251204T124815
```
This will fail with `No variable found with this name: stackhpc_pulp_repo_doca_2_9_1_rhel9_7_version`, see https://github.com/stackhpc/stackhpc-kayobe-config/blob/stackhpc/2025.1/etc/kayobe/ofed.yml#L15

elelaysh replied:
thank you Piotr. I'll provide a fix shortly

elelaysh replied:
see `doca_distribution` in `etc/kayobe/ofed.yml`

cityofships replied:
Looks like we're still missing the modules:
```
No variable found with this name: stackhpc_pulp_repo_doca_3_2_1_rhel9_modules_version
```

priteau replied:
@cityofships Can you check if it is better now?

cityofships replied:
The baseurl is set to something [different](https://github.com/stackhpc/stackhpc-release-train/pull/457/changes#diff-79b98dc6a02dff7f0a526062916901d18f3002f6aa8d5f274aae46483f75f40eR899):
```
FAILED - RETRYING: [localhost]: Sync RPM remotes into repositories (3 retries left).
FAILED - RETRYING: [localhost]: Sync RPM remotes into repositories (2 retries left).
FAILED - RETRYING: [localhost]: Sync RPM remotes into repositories (1 retries left).
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: Exception: Task failed to complete. (failed; 404, message='Not Found', url=URL('https://ark.stackhpc.com/pulp/content/doca/3.2.1/rhel9.7/x86_64/20251219T132023'))
failed: [localhost] (item=DOCA Online Repo 3.2.1 - RHEL 9.7) =>
    ansible_index_var: repository_index
    ansible_loop_var: item
    attempts: 3
    changed: false
    item: DOCA Online Repo 3.2.1 - RHEL 9.7
    msg: Task failed to complete. (failed; 404, message='Not Found', url=URL('https://ark.stackhpc.com/pulp/content/doca/3.2.1/rhel9.7/x86_64/20251219T132023'))
    repository_index: 9
```
After fixing it:
```
Error:
 Problem 1: package iser-dkms-25.10-OFED.25.10.1.7.1.1.el9.x86_64 from doca obsoletes kmod-iser < 25.10-OFED.25.10.1.7.1.1.el9 provided by kmod-iser-24.10-OFED.24.10.3.2.5.1.1.x86_64 from @System
  - package doca-ofed-3.2.1-044000.x86_64 from doca requires iser-dkms >= 25.10-OFED.25.10.1.7.1.1.el9, but none of the providers can be installed
  - cannot install the best update candidate for package kmod-iser-24.10-OFED.24.10.3.2.5.1.1.x86_64
  - cannot install the best update candidate for package doca-ofed-2.9.3-0.2.2.x86_64
 Problem 2: problem with installed package doca-ofed-2.9.3-0.2.2.x86_64
  - package doca-ofed-3.2.1-044000.x86_64 from doca requires isert-dkms >= 25.10-OFED.25.10.1.7.1.1.el9, but none of the providers can be installed
  - package doca-ofed-2.9.3-0.2.2.x86_64 from @System requires hcoll >= 4.8.3230-1.2410068, but none of the providers can be installed
  - package isert-dkms-25.10-OFED.25.10.1.7.1.1.el9.x86_64 from doca obsoletes kmod-isert < 25.10-OFED.25.10.1.7.1.1.el9 provided by kmod-isert-24.10-OFED.24.10.3.2.5.1.1.x86_64 from @System
  - package doca-ofed-userspace-3.2.1-044000.x86_64 from doca conflicts with hcoll < 4.8.3233 provided by hcoll-4.8.3230-1.2410068.x86_64 from @System
  - package doca-ofed-userspace-3.2.1-044000.x86_64 from doca obsoletes hcoll < 4.8.3233 provided by hcoll-4.8.3230-1.2410068.x86_64 from @System
  - cannot install the best update candidate for package kmod-isert-24.10-OFED.24.10.3.2.5.1.1.x86_64
  - cannot install the best update candidate for package hcoll-4.8.3230-1.2410068.x86_64
(try to add '--allowerasing' to command line to replace conflicting packages or '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```
Maybe it's how DOCA installation works? Anyway, with `--nobest`:
```
Package doca-kernel-repo-25.10.1.7.1.0-1.kver.5.14.0.611.13.1.el9.7.x86.64.x86_64.rpm is not signed
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: GPG check FAILED
```

elelaysh replied:
Starting from a fresh multinode install
 1. an issue with `/etc/yum.repos.d/doca.repo` using _development_ instead of the date _20251222T160623_ in the baseurl. This is fixed by setting `stackhpc_repo_rhel9_doca_version: 20251219T132023` and `stackhpc_repo_rhel9_doca_modules_version` in `etc/kayobe/stackhpc.yml`
 2. an issue where gpgcheck=1 in `/etc/yum.repos.d/doca.repo`, where we need it to 0 because we don't give the gpgkey url in `etc/kayobe/dnf.yml`. This is fixed by running `kayobe overcloud host configure -t dnf`
 3. then I had no issue with `kayobe overcloud host package update --packages "*" --limit mlnx`
 4. and no issue with `kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/tools/install-doca.yml`

elelaysh replied:
@cityofships from provided error message this seems like an upgrade issue with installed doca-ofed-2.9.3. How can I reproduce it?

elelaysh replied:
has been discussed elsewhere: a local pulp server was in place. Fixes have been pushed to address the issues when using a local pulp server.

cityofships commented on etc/kayobe/pulp-repo-versions.yml:
```diff
@@ -76,6 +77,21 @@ stackhpc_pulp_repo_rocky_9_6_extras_version: 20250726T040613
 stackhpc_pulp_repo_rocky_9_6_extras_aarch64_version: 20250726T045704
 stackhpc_pulp_repo_rocky_9_6_highavailability_version: 20250605T150141
 stackhpc_pulp_repo_rocky_9_6_highavailability_aarch64_version: 20250605T150141
+stackhpc_pulp_repo_rocky_9_7_appstream_version: 20251203T103807
```
Repositories not yet promoted?
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: Exception: Task failed to complete. (failed; 403, message='Forbidden', url=URL('https://ark.stackhpc.com/pulp/content/rocky/9.7/AppStream/x86_64/os/20251203T103807'))
failed: [localhost] (item=Rocky Linux 9 - AppStream) =>
    ansible_index_var: repository_index
    ansible_loop_var: item
    attempts: 3
    changed: false
    item: Rocky Linux 9 - AppStream
    msg: Task failed to complete. (failed; 403, message='Forbidden', url=URL('https://ark.stackhpc.com/pulp/content/rocky/9.7/AppStream/x86_64/os/20251203T103807'))
    repository_index: 0
```

priteau replied:
Not yet, do you need them for a customer?

Alex-Welsh commented on etc/kayobe/kolla/kolla-build.conf:
```diff
@@ -7,7 +7,7 @@ base_tag = noble-20241118.1
 {# Similarly pinning to Rocky 9 minor version used in our repos #}
 {% elif kolla_base_distro == 'rocky' %}
-base_tag = 9.{{ stackhpc_pulp_repo_rocky_9_minor_version }}
+base_tag = 9
```
What's the reason for this change?

elelaysh replied:
the rockylinux 9.7 tag was not yet available in quay.io.
It is now. I've removed this hack

Alex-Welsh commented on etc/kayobe/kolla/kolla-build.conf:
```diff
@@ -16,7 +16,7 @@ build_args = {{ (kolla_build_args | default({})).items() | map('join', ':') | jo
 type = git
 location = https://github.com/stackhpc/requirements
-reference = stackhpc/{{ openstack_release }}
+reference = bump-django-4.2
```
It would be best to get this merged into one of our main branches if we're building our images from it

elelaysh replied:
yes, totally agree

elelaysh replied:
merged

Alex-Welsh commented on etc/kayobe/kolla-image-tags.yml:
```diff
@@ -4,25 +4,25 @@
     ubuntu-noble: 2025.1-ubuntu-noble-20250822T151934
   rabbitmq:
-    rocky-9: 2025.1-rocky-9-20250905T151507
+    rocky-9: 2025.1-rocky-9-20251208T092405
```
the `openstack:` tag is the default tag for each OS. If the tags are identical, that is the only one that needs to be set, and all the others can be removed:


```suggestion
    ubuntu-noble: 2025.1-ubuntu-noble-20250730T105631
  iscsid:
    ubuntu-noble: 2025.1-ubuntu-noble-20251013T135156
  keystone:
    ubuntu-noble: 2025.1-ubuntu-noble-20251031T094539
  kolla_toolbox:
    ubuntu-noble: 2025.1-ubuntu-noble-20250905T151507
  magnum:
    ubuntu-noble: 2025.1-ubuntu-noble-20251016T073648
  prometheus_libvirt_exporter:
    ubuntu-noble: 2025.1-ubuntu-noble-20250822T151934
  rabbitmq:
```

(I'm not sure if I got the spacing right in the suggestion above, the GitHub UI is not my text editor of choice)

elelaysh replied:
thanks for the tip Alex. I will do container images in a separate PR

Alex-Welsh commented on releasenotes/notes/rl9.7-c8d033f11097b76b.yaml:
```diff
@@ -0,0 +1,9 @@
+    XXXX PENDING CONTAINER BUILD
+upgrade:
+  - |
+    9.7 is now the default release for Rocky Linux.
```
I know we have done it in the past, but I don't think we need and upgrade note for this. It fits better in the "features".

The "upgrade" section should only be used if there is some sort of breaking change when users upgrade to the given version

elelaysh replied:
ok, will put both in features. Thanks Alex.

Alex-Welsh commented on etc/kayobe/pulp-repo-versions.yml:
We should also bump the 3rd party repos used in container image builds for Rocky images (Grafana, Opensearch etc)

elelaysh replied:
will do container images in a separate PR

elelaysh commented:
Succesful test: https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/20235991635

elelaysh commented:
Build overcloud host image: https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/20226203029

=> image contains rocky repos

elelaysh commented:
Tempest passed on small test clusters in OVN and OVS mode

OVN
```
Ran: 334 tests in 907.182 sec.
 - Success: 258
 - Skipped: 76
 - Expected failures: 0
 - Unexpected success: 0
 - Failures: 0
```

OVS
```
Ran: 334 tests in 666.650 sec.
 - Success: 258
 - Skipped: 76
 - Expected failures: 0
 - Unexpected success: 0
 - Failures: 0
```

priteau commented on etc/kayobe/pulp.yml:
```diff
@@ -200,8 +200,8 @@ stackhpc_pulp_distribution_deb_production: >-
 stackhpc_pulp_sync_rocky_9: "{{ os_distribution == 'rocky' }}"
-# Rocky 9 minor version number. Supported values: 6. Default is 6
-stackhpc_pulp_repo_rocky_9_minor_version: '6'
+# Rocky 9 minor version number. Supported values: 7. Default is 7
```
@elelaysh Please keep 6 as one of the supported values, as suggested here.

priteau commented on releasenotes/notes/rl9.7-c8d033f11097b76b.yaml:
```diff
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    Added support for Rocky Linux 9.7 (container image refresh in separate PR)
```
These release notes are meant for consumption outside of GitHub so it is better to leave out development details. Just keep:

Added support for Rocky Linux 9.7.

elelaysh replied:
done, thanks

priteau reviewed (dismissed):
LGTM, thanks @elelaysh

priteau reviewed (commented):
We need to build and push DOCA modules.

priteau committed:
Add DOCA 3.2.1 modules

Remove DOCA 3.2.0: a single version is enough for RL 9.7

cityofships commented on etc/kayobe/pulp-repo-versions.yml:
```diff
@@ -18,6 +18,9 @@ stackhpc_pulp_repo_doca_2_9_1_rhel9_5_version: 20241211T171301
 stackhpc_pulp_repo_doca_2_9_3_rhel9_6_modules_version: 20250714T141841
+stackhpc_pulp_repo_doca_3_2_0_rhel9_version: 20251204T124815
+stackhpc_pulp_repo_doca_3_2_1_rhel9_version: 20251219T132023
+stackhpc_pulp_repo_doca_3_2_1_rhel9_7_modules_version: 20251222T160623
```
Are we expecting to create a new variable with every minor release?

elelaysh replied:
You're right: it should be one version per RL version. I'ved removed 3_2_0

Alex-Welsh commented on etc/kayobe/ofed.yml:
```diff
@@ -12,7 +15,8 @@ stackhpc_pulp_doca_version: "{{ '2.9.3' if stackhpc_pulp_repo_rocky_9_minor_vers
 
 # DOCA Snapshot lookup vars
-doca_version_lookup_var: "stackhpc_pulp_repo_doca_{{ stackhpc_pulp_doca_version | replace('.', '_') }}_rhel9_{{ stackhpc_pulp_repo_rocky_9_minor_version }}_version"
+doca_distribution: "{% if stackhpc_pulp_doca_version is version('3.2.0', '>=') %}rhel9{% else %}rhel9_{{ stackhpc_pulp_repo_rocky_9_minor_version }}{% endif %}"
```
Could you just use `stackhpc_doca_distribution` instead?

elelaysh replied:
yes, please see newer commit b539da32.
There was a difference of dot vs underscore. Now made more obvious by the name `doca_rocky_9_minor_dot` hopefully.

Alex-Welsh commented on releasenotes/notes/rl9.7-c8d033f11097b76b.yaml:
```diff
@@ -0,0 +1,6 @@
+---
+features:
```
Probably worth mentioning doca 3.2.1 as well

Alex-Welsh commented on etc/kayobe/stackhpc.yml:
```diff
@@ -49,7 +49,8 @@ stackhpc_repo_docker_ce_ubuntu_noble_version: "{{ stackhpc_repo_distribution }}"
 # DOCA
-stackhpc_repo_rhel9_doca_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/doca/{{ stackhpc_pulp_doca_version }}/rhel9.{{ stackhpc_pulp_repo_rocky_9_minor_version }}/x86_64/{{ stackhpc_repo_rhel9_doca_version }}"
+stackhpc_doca_distribution: "{% if stackhpc_pulp_doca_version is version('3.2.0', '>=') %}rhel9{% else %}rhel9.{{ stackhpc_pulp_repo_rocky_9_minor_version }}{% endif %}"
+stackhpc_repo_rhel9_doca_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/doca/{{ stackhpc_pulp_doca_version }}/{{ stackhpc_doca_distribution }}/x86_64/{{ stackhpc_repo_rhel9_doca_version }}"
```
What actually happened here that requires this change? I don't think it's going to work

elelaysh replied:
doca before 3.2.0 had the rhel9 minor version version in the url: https://linux.mellanox.com/public/repo/doca/2.9.3/rhel9.6/

It's not the case anymore: https://linux.mellanox.com/public/repo/doca/3.2.1/rhel9/

cityofships commented:
```
msg: Task failed to complete. (failed; 404, message='Not Found', url=URL('https://ark.stackhpc.com/pulp/content/doca/3.2.1/rhel9.7/x86_64/20251219T132023'))
```

elelaysh committed:
[skip ci] Fix DOCA url for local pulp servers

removes the RockyLinux minor version in the name and path
when DOCA version is greater than 3.2.0.
Doesn't apply to DOCA modules because they are still compiled
for a specific RL minor version.

elelaysh committed:
Bump all package version for RockyLinux

cityofships commented:
Rebased

cityofships reviewed (dismissed):
LGTM

elelaysh committed:
stackhpc-cloud-tests: bump docker minmax versions

Latest version for RockyLinux is 29.1.4,
Ubuntu noble has 27.5.1.

elelaysh committed:
Add playbook to automate Rocky 9.7 with OFED upgrade

Tested on multinode.

Fix install-doca.yml to not install doca-ofed anymore (avoid dkms).

The stackhpc_doca_kernel_version_matrix variable contains kernel module
versions to install for last 2 supported minor RockyLinux versions.
It must be changed after a new pre-compiled kernel module version has been built.

priteau commented on etc/kayobe/ansible/maintenance/rocky-97-ofed-upgrade.yml:
```diff
@@ -0,0 +1,89 @@
+
+    - name: Ensure appropriate doca-kernel is installed
+      ansible.builtin.dnf:
+        name: doca-kernel-5.14.0.611.13.1.el9.7.x86.64
```
I suggest putting this in a variable that can be overridden.

elelaysh replied:
playbook variable, taking its value from `etc/kayobe/ofed.yml` by default.

priteau commented on etc/kayobe/ansible/maintenance/rocky-97-ofed-upgrade.yml:
```diff
@@ -0,0 +1,89 @@
+    - name: Reboot to apply updates
+      ansible.builtin.reboot:
+        reboot_timeout: "{{ reboot_timeout_s }}"
+        connect_timeout: 600
```
Add back search_paths like in ubuntu-upgrade.yml or reuse reboot.yml

elelaysh replied:
done

priteau commented on etc/kayobe/ansible/maintenance/rocky-97-ofed-upgrade.yml:
```diff
@@ -0,0 +1,89 @@
+# ANSIBLE_ROLES_PATH is defined and includes '$KAYOBE_PATH/ansible/roles' on the Ansible control host.
+# Where KAYOBE_PATH is the path to the source of kayobe that the environment uses.
+- name: Prepare upgrade from Rocky Linux 9.6 to 9.7
+  hosts: mlnx
```
Add way to configure serial or reuse reboot.yml

elelaysh replied:
done

priteau commented on etc/kayobe/ansible/maintenance/rocky-97-ofed-upgrade.yml:
```diff
@@ -0,0 +1,89 @@
+      become: true
+
+- name: Confirm the host is upgraded to Rocky Linux 9.7
+  hosts: overcloud:infra-vms:seed:seed-hypervisor
```
Change limit to mlnx

elelaysh replied:
done

priteau commented on etc/kayobe/ansible/maintenance/rocky-97-ofed-upgrade.yml:
```diff
@@ -0,0 +1,89 @@
+- name: Confirm the host is upgraded to Rocky Linux 9.7
+  hosts: overcloud:infra-vms:seed:seed-hypervisor
+  vars:
+    ansible_python_interpreter: /usr/bin/python3
```
Can be removed because we are not upgrading Python.

elelaysh replied:
done

priteau commented on etc/kayobe/ansible/maintenance/rocky-97-ofed-upgrade.yml:
```diff
@@ -0,0 +1,89 @@
+      ansible.builtin.assert:
+        that:
+          - ansible_facts.distribution == 'Rocky'
+          - ansible_facts.distribution_version == '9.6'
```
s/9.6/9.7/

elelaysh replied:
done

priteau commented on etc/kayobe/ansible/maintenance/rocky-97-ofed-upgrade.yml:
```diff
@@ -0,0 +1,89 @@
+        name: "*"
+        state: latest
+      become: true
+
```
Consider running reset-bls-entries.yml here.

elelaysh replied:
done

priteau commented on etc/kayobe/ansible/maintenance/rocky-97-ofed-upgrade.yml:
```diff
@@ -0,0 +1,89 @@
+---
+# To prevent Ansible role dependency errors, this playbook requires that environment variable
+# ANSIBLE_ROLES_PATH is defined and includes '$KAYOBE_PATH/ansible/roles' on the Ansible control host.
+# Where KAYOBE_PATH is the path to the source of kayobe that the environment uses.
```
Unsure if this comment is relevant.

elelaysh replied:
removed the comment

elelaysh commented:
I need to test again the upgrade playbook

mnasiadka commented:
Can we break out DOCA from this PR to a separate one? I don't think we should be delaying the 9.7 availability based on some additional DOCA tests.

elelaysh commented:
@mnasiadka the upgrade and install playbooks have been tested this morning. Maybe we can review one final time?

Alex-Welsh commented on releasenotes/notes/rl9.7-c8d033f11097b76b.yaml:
```diff
@@ -0,0 +1,8 @@
+    Added support for Rocky Linux 9.7.
+  - |
+    9.7 is now the default release for Rocky Linux.
+  - |
+    The upgrade to DOCA 3.2.1 is required for Rocky Linux 9.7.
```
nit: should be an `upgrade` note, not feature.

Feature note would be `New thing is now supported`

Upgrade note would be `When you upgrade to this release you have to be aware that this thing is happening`

Not worth holding up this PR for, but if you need to make other changes, might be worth bundling this one in too

elelaysh replied:
yes, thank you for the explanation.

I also would like to compact the commit history

Alex-Welsh replied:
We can always squash & merge if required

elelaysh replied:
I've applied your suggestions, rebased and merged relevant commits.

Alex-Welsh commented on etc/kayobe/ofed.yml:
```diff
@@ -3,7 +3,15 @@
+stackhpc_pulp_doca_version: "{{ stackhpc_pulp_doca_version_matrix[stackhpc_pulp_repo_rocky_9_minor_version] | default('2.9.1') }}"
+
+# Available and tested versions of the pre-compiled doca-ofed kernel modules
+stackhpc_doca_kernel_version:
```
nit: by your own convention, this should have the `_matrix` suffix
```suggestion
stackhpc_doca_kernel_version_matrix:
```

Again, not worth holding up the PR for but maybe worth tweaking if another change _is_ required

elelaysh replied:
will do

Alex-Welsh reviewed (commented):
A couple of nitpicks but generally I think the change is good to go now. If it passes CI I think we just merge it as-is

elelaysh commented:
thanks for the merge :+1:
